### DISCUSSION TITLE
Latest optimal values for teaching a face?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [TheLastBen/fast-stable-diffusion#1667] [Q&A]
first question

I am teaching a face

naming images as "ohwx man" right?

then

using concept images generated by "photo of man" ?

how many concept images do you suggest?

when using Offset_Noise

how many UNet_Training_Steps per training image?

is this UNet_Learning_Rate= "2e-6" good?

how many Text_Encoder_Training_Steps per training image?

Text_Encoder_Learning_Rate= "1e-6" is this good?

what is this Text_Encoder_Concept_Training_Steps?

this means that model is doing extra training on concept images? if this is 0, concept images never used during training?

**Comment by TheLastBen** [comment]
For naming, use only the token "ohwx", don't use any other word and make sure the token is unknown by the model.

Do not use concept images for a face, concept training will just parasite the training.

Try 150 UNet steps per image, then resume the training for more if the result isn't enough keep the lr at 2e-6, keep the number of images below 20

for the text encoder keep the total steps to 350, if you resume training set it to 0 as it is already trained. lr 1e-6

if you set any steps to 0, it will not train with that specific process (eg; Text_Encoder_Concept_Training_Steps)

**Reply by FurkanGozukara** [reply]
awesome answers ty so much. 

so what is being Text_Encoder_Concept_Training_Steps ? it is training concept images? 

could you give some more information regarding style training

when do we need concept images?

i am planning to make a video hopefully for your repo but trying to understand some fundamental things first. I have got some knowledge for automatic1111 dreambooth extension

**Reply by TheLastBen** [reply]
Concept images serve as class images for regularization, so you pick them just like you pick reg images, but you keep the number of text encoder steps low, you don't need concept images, personally I've never used them for faces or for styles.

try this dataset for a style, they are from midjourney, the difference is noticeable with and without offset noise 
[froggy.zip](https://github.com/TheLastBen/fast-stable-diffusion/files/10874528/froggy.zip)
keep the token as "daiton", at inference, use only "daiton" as the prompt for testing
